aeron-cluster/src/main/java/io/aeron/cluster/ConsensusModuleAgent.java

```diff
@@ -2604,8 +2607,13 @@ private int processPendingSessions(
                 {
                     case CLIENT:
                     {
+                        if (Cluster.Role.LEADER == role)
+                        {
+                            nextCommittedSessionId = Math.max(session.id() + 1, nextCommittedSessionId);
+                        }
```

Comment:
`processPendingSessions` method is only called when `role == LEADER` so no need to check for the role.
However, there is another issue, i.e. update of `nextCommittedSessionId` before session is appended to the log. Instead the logic should be moved inside of the if statement body for `session.appendSessionToLogAndSendOpen` and it should be simplified to:
```java
nextCommittedSessionId = session.id() + 1;
```